### PR TITLE
Implement monitor mode

### PR DESCRIPTION
**Pull request by ereslibre**
Depends on: https://github.com/kubewarden/policy-server/pull/194
Fixes: https://github.com/kubewarden/policy-server/issues/190

### CODE REVIEW

**Review by raulcabello — CHANGES_REQUESTED** [review]
LGTM, but I think you forgot to push the crd changes for the policy mode. They are automatically generated when you run make deploy, but it'd be great if you push them here.
Also don't forget to update the crd in the helm chart https://github.com/kubewarden/helm-charts/tree/main/charts/kubewarden-crds

**Comment by ereslibre** [comment]
> LGTM, but I think you forgot to push the crd changes for the policy mode. They are automatically generated when you run make deploy, but it'd be great if you push them here.

Right!, updated

**Comment by ereslibre** [comment]
Done in https://github.com/kubewarden/helm-charts/pull/63, it will need a version bump but I can do that outside of the PR.

**Review by raulcabello — APPROVED** [review]

**Comment by raulcabello** [comment]
Thanks @ereslibre!

**Review by jvanz — APPROVED** [review]
LGTM! :D

**Review by flavio — APPROVED** [review]
LGTM, I just left a question out of curiosity

File: config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml [hunk 1]
```diff
@@ -74,6 +74,18 @@ spec:
...
+                default: protect
+                description: Mode defines the execution mode of this policy. Can be
+                  set to either "protect" or "monitor". If it's empty, it is defaulted
+                  to "protect". Transitioning this setting from "monitor" to "protect"
+                  is allowed, but is disallowed to transition from "protect" to "monitor".
+                  To perform this transition, the policy should be recreated in "monitor"
+                  mode instead.
+                enum:
+                - protect
+                - monitor
```
[the remark this answers is not in the document]
Comment (ereslibre) [reply config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml:87]: Yes, they can only provide the values specified in this list, gated by the API server itself.

**Pull request merged**